doc/src/manual/variables-and-scoping.md

```diff
@@ -1,33 +1,36 @@
 # [Scope of Variables](@id scope-of-variables)
 
 The *scope* of a variable is the region of code within which a variable is visible. Variable scoping
-helps avoid variable naming conflicts. The concept is intuitive: two functions can both have arguments
-called `x` without the two `x`'s referring to the same thing. Similarly, there are many other cases
-where different blocks of code can use the same name without referring to the same thing. The
-rules for when the same variable name does or doesn't refer to the same thing are called scope
-rules; this section spells them out in detail.
+helps avoid variable naming conflicts. The concept is intuitive: two functions can both have
+arguments called `x` without the two `x`'s referring to the same thing. Similarly, there are many
+other cases where different blocks of code can use the same name without referring to the same
+thing. The rules for when the same variable name does or doesn't refer to the same thing are called
+scope rules; this section spells them out in detail.
 
 Certain constructs in the language introduce *scope blocks*, which are regions of code that are
 eligible to be the scope of some set of variables. The scope of a variable cannot be an arbitrary
-set of source lines; instead, it will always line up with one of these blocks. There are two
-main types of scopes in Julia, *global scope* and *local scope*. The latter can be nested. The
-constructs introducing scope blocks are:
+set of source lines; instead, it will always line up with one of these blocks. There are two main
+types of scopes in Julia, *global scope* and *local scope*. The latter can be nested. There is also
+a distinction in Julia between constructs which introduce a "hard scope" and those which only
+introduce a "soft scope", which affects whether shadowing a global variable by the same name is
+allowed or not.
 
 ### [Scope constructs](@id man-scope-table)
 
-Construct | Scope type | Scope blocks it may be nested in
------------- | -------------  |---------------------------
-[`module`](@ref), [`baremodule`](@ref)            | global | global
-interactive prompt (REPL)                         | global | global
-(mutable) [`struct`](@ref), [`macro`](@ref)       | local  | global
-[`for`](@ref), [`while`](@ref), [`try-catch-finally`](@ref try), [`let`](@ref) |local | global or local
-functions (either syntax, anonymous & do-blocks) | local | global or local
-comprehensions, broadcast-fusing                 | local | global or local
+The constructs introducing scope blocks are:
+
+Construct | Scope type | Allowed within
+--------- | ---------  |---------------
+files, REPL (i.e. `Main` module), [`module`](@ref), [`baremodule`](@ref) | global | global
```

Comment:
I would not mention files or the REPL here, since neither delimit scope blocks. A file doesn't get its own scope.

Reply:
Fair point, it was in there before so I left it, but happy to delete.